### PR TITLE
[WIP] feat(lint): noAccessKey for HTML & JSX

### PR DESCRIPTION
**Pull request by mainangethe**
## Summary

Implements #1345 

Ported the rule `noAccessKey` to the new category `a11y`
- Migrated the JXS linting rules
-  Added HTML rules

## Test Plan

- Migrated the existing JSX tests
- Added HTML tests
- Confirmed `./rome tests & ./rome ci` passes

### CODE REVIEW

File: internal/compiler/lint/rules/a11y/noAccessKey.test.rjson [hunk 1]
```diff
@@ -0,0 +1,8 @@
+filename: "file.html"
+invalid: [
+	"<input accesskey='key />"
```
Comment (ematipico) [inline internal/compiler/lint/rules/a11y/noAccessKey.test.rjson:3]: This is not valid HTML. If you check the snapshot, there's a parse error. We should just remove it, it's not needed.

**Review by ematipico — CHANGES_REQUESTED** [review]
Looks good. Just one small thing

File: internal/compiler/lint/rules/a11y/noAccessKey.test.rjson (hunk 1 above)
Comment (mainangethe) [reply internal/compiler/lint/rules/a11y/noAccessKey.test.rjson]: Kept it in the JSX side, removed it in the HTML side of things..

**Comment by ematipico** [comment]
Hi @mainangethe , any chance to get this merged?

**Comment by ematipico** [comment]
@mainangethe I am going to close this PR in favour of #1442 

I cherry picked your commits, so your changes are still visible.

**Pull request closed without merging**